### PR TITLE
conda package walkthrough

### PR DESCRIPTION
**Pull request by rrigdon**
Added Sophia C.'s conda package walkthrough to the conda-build docs

### CODE REVIEW

File: docs/source/conda-package-walkthrough/index.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+.. toctree::
+   :maxdepth: 1
+
+   overview
+   package_anatomy
+   generating_index
+   conda_install
+   building_anaconda
+
+.. Indices and tables
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/index.rst]: I think these lines (from 17-22) are commented out, they can be deleted

File: docs/source/conda-package-walkthrough/package_anatomy.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  ├── bin
+  │   └── pyflakes
+  ├── info
+  │   ├── LICENSE.txt
+  │   ├── files
+  │   ├── index.json
+  │   ├── paths.json
+  │   └── recipe
+  └── lib
+    └── python3.5
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/package_anatomy.rst]: this line needs a couple more spaces to be in line with the ones above it. eg.

```
  │   ├── files
  │   ├── index.json
  │   ├── paths.json
  │   └── recipe
  └── lib
      └── python3.5
```

File: docs/source/conda-package-walkthrough/package_anatomy.rst [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+* index.json
+
+  * metadata about the package including platform, version, dependencies, build info
+
+* paths.json
+
+  * a list of files in the package, along with their associated SHA-256, size in bytes,
+    and the type of path (eg. hardlink vs. softlink)
+
+index.json
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/package_anatomy.rst]: I think this can go under the index.json bullet point. Initially I put it here because it worked better for slides

File: docs/source/conda-package-walkthrough/package_anatomy.rst [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    ],
+    "license": "BSD 3-Clause",
+    "name": "fun-packge",
+    "platform": "linux",
+    "subdir": "linux-64",
+    "timestamp": 1535416612069,
+    "version": "0.0.0"
+  }
+
+paths.json
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/package_anatomy.rst]: same thing for this one, I think it can go under the paths.json bullet point

File: docs/source/conda-package-walkthrough/generating_index.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  * Remove all packages that need to be removed
+
+  * For all packages that need to be added/updated
+
+    * Extract the package to access metadata including full package name,
+      mtime, size, index.json
+
+    * Add package to repodata
+
+EXAMPLE: Building a channel
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/generating_index.rst]: I think "EXAMPLE" looks better as "Example"

File: docs/source/conda-package-walkthrough/conda_install.rst [hunk 6]
```diff
@@ -0,0 +1,54 @@
+*********************
+Installing with conda
+*********************
+
+Conda install
+-------------
+
+* When you `conda install <package>` the happy path (with no dependencies)
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/conda_install.rst]: I think it would be neat if this turns into a flow chart or something at some point and also includes not just the "happy path". Maybe that can be for another PR though?

File: docs/source/conda-package-walkthrough/generating_index.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+----------------------
```
Comment (soapy1) [inline docs/source/conda-package-walkthrough/generating_index.rst]: This `---` line should go the whole length of the title e.g

```
EXAMPLE: Building a channel
----------------------------
``

File: docs/source/conda-package-walkthrough/generating_index.rst (hunk 5 above)
Comment (rrigdon) [reply docs/source/conda-package-walkthrough/generating_index.rst]: Our existing style guide uses all caps for notes and examples so I'm trying to keep it consistent with the AE and other docs.

File: docs/source/conda-package-walkthrough/conda_install.rst (hunk 6 above)
Comment (rrigdon) [reply docs/source/conda-package-walkthrough/conda_install.rst]: Agreed that this would be a nice option for another PR. I made at ticket for it: https://anaconda.atlassian.net/browse/DSNC-972

File: docs/source/conda-package-walkthrough/generating_index.rst (hunk 5 above)
Comment (soapy1) [reply docs/source/conda-package-walkthrough/generating_index.rst]: Sounds good to me!

File: docs/source/conda-package-walkthrough/index.rst [hunk 8]
```diff
@@ -0,0 +1,22 @@
+***********************************************************
+Conda packages walkthrough and building Anaconda installers
+***********************************************************
+
+Topics
+------
```
[the remark this answers is not in the document]
Comment (rrigdon) [reply docs/source/conda-package-walkthrough/index.rst]: Agreed - hopefully I did it right!
